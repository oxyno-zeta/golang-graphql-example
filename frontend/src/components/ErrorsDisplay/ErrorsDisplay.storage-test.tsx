import { CombinedGraphQLErrors, type ServerError } from '@apollo/client';
import { GraphQLError } from 'graphql';

// Build data for tests
export const forbiddenNetworkError: ServerError = {
  name: 'Forbidden',
  message: 'Forbidden',
  response: new Response(),
  statusCode: 403,
  bodyText: '',
};
export const simpleGraphqlErrorWithoutExtension: GraphQLError = new GraphQLError('simple graphql error');
export const simpleCombinedGraphQLErrorWithoutExtension = new CombinedGraphQLErrors({
  errors: [simpleGraphqlErrorWithoutExtension],
});
export const simpleForbiddenGraphqlError: GraphQLError = new GraphQLError('forbidden graphql error', {
  extensions: { code: 'FORBIDDEN' },
});
export const simpleForbiddenCombinedGraphQLError = new CombinedGraphQLErrors({ errors: [simpleForbiddenGraphqlError] });
export const simpleInternalServerErrorGraphqlError: GraphQLError = new GraphQLError(
  'internal server error graphql error',
  {
    extensions: { code: 'INTERNAL_SERVER_ERROR' },
  },
);
export const simpleInternalServerErrorCombinedGraphQLError = new CombinedGraphQLErrors({
  errors: [simpleInternalServerErrorGraphqlError],
});
