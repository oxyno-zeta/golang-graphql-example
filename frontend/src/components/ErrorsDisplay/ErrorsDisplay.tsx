import React, { Key, ReactNode } from 'react';
import Typography, { TypographyProps } from '@mui/material/Typography';
import Box from '@mui/material//Box';
import { ApolloError, ServerError } from '@apollo/client';
import { useTranslation } from 'react-i18next';
import type { SxProps } from '@mui/material';

export interface Props {
  error?: ApolloError | Error | null;
  errors?: (ApolloError | Error)[];
  noMargin?: boolean;
  containerBoxSx?: SxProps;
  errorTitleTypographyProps?: TypographyProps;
  errorElementTypographyProps?: TypographyProps;
  ulSx?: SxProps;
  liSx?: SxProps;
}

export interface CustomNetworkError {
  message: string;
  path: null | undefined | ReadonlyArray<string | number>;
}

const defaultProps = {
  error: null,
  errors: [],
  noMargin: false,
  containerBoxSx: {},
  errorTitleTypographyProps: {},
  errorElementTypographyProps: {},
  ulSx: {},
  liSx: {},
};

/* eslint-disable react/no-array-index-key */
function ErrorsDisplay({
  error,
  errors,
  noMargin,
  containerBoxSx,
  errorTitleTypographyProps,
  errorElementTypographyProps,
  ulSx,
  liSx,
}: Props) {
  // Initialize translate
  const { t } = useTranslation();

  // Check if error or errors are set
  if ((!error || !errors) && errors && errors.length === 0) {
    // Display nothing
    return null;
  }

  // Build error list
  let errorList = errors;
  // Check if error is set
  if (error) {
    // Override array
    errorList = [error];
  }

  const elements: ReactNode[] = [];

  // Loop over errors
  errorList?.forEach((err, mainIndex) => {
    const contents: { content: string; key: Key }[] = [];
    if ((err as ApolloError).networkError || (err as ApolloError).graphQLErrors) {
      // Create intermediate variable just for typescript.............
      const workingErr: ApolloError = err as ApolloError;

      if (
        workingErr.networkError &&
        (!(workingErr.networkError as ServerError).result ||
          Object.keys((workingErr.networkError as ServerError).result).length === 0 ||
          !workingErr.graphQLErrors)
      ) {
        contents.push({ content: workingErr.networkError.message, key: mainIndex });
      }

      if (workingErr.graphQLErrors) {
        contents.push(
          ...workingErr.graphQLErrors.map(({ message, extensions }, i) => {
            let mess = message;
            // Check if there is a code in extensions
            if (extensions && extensions.code) {
              mess = t(`common.errorCode.${extensions.code}`);
            }

            return { content: mess, key: `${mainIndex}-graphQLErrors-${i}` };
          }),
        );
      }

      if (
        workingErr.networkError &&
        (workingErr.networkError as ServerError).result &&
        (workingErr.networkError as ServerError).result.errors
      ) {
        contents.push(
          ...((workingErr.networkError as ServerError).result.errors as [CustomNetworkError]).map(
            ({ message, path }, i) => ({
              content: `${path?.join('.')} ${message}`,
              key: `${mainIndex}-networkError-${i}`,
            }),
          ),
        );
      }
    } else {
      // Isn't an apollo error
      contents.push({ content: err.message, key: mainIndex });
    }

    if (contents) {
      // Check if a content have been detected
      // Save element
      elements.push(
        ...contents.map(({ content, key }) => (
          <Box component="li" key={key} sx={liSx}>
            <Typography color="error" {...errorElementTypographyProps}>
              {content}
            </Typography>
          </Box>
        )),
      );
    }
  });

  return (
    <Box sx={{ color: 'error', margin: noMargin ? 0 : 8, ...containerBoxSx }}>
      <Typography color="error" {...errorTitleTypographyProps}>
        {t('common.errors')}:
      </Typography>
      <Box component="ul" sx={ulSx}>
        {elements}
      </Box>
    </Box>
  );
}
/* eslint-enable */

ErrorsDisplay.defaultProps = defaultProps;

export default ErrorsDisplay;
