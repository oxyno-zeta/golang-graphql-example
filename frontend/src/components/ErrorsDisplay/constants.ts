export const ServerErrorCustomComponentMapKey = 'SERVER_ERROR';

export const GraphqlErrorsExtensionsCodeCustomComponentMapKeyPrefix = 'GRAPHQL_ERRORS_EXTENSIONS_CODE_';

export const GraphqlErrorsExtensionsCodeConflictCustomComponentMapKey = `${GraphqlErrorsExtensionsCodeCustomComponentMapKeyPrefix}CONFLICT`;
export const GraphqlErrorsExtensionsCodeForbiddenCustomComponentMapKey = `${GraphqlErrorsExtensionsCodeCustomComponentMapKeyPrefix}FORBIDDEN`;
export const GraphqlErrorsExtensionsCodeInternalServerErrorCustomComponentMapKey = `${GraphqlErrorsExtensionsCodeCustomComponentMapKeyPrefix}INTERNAL_SERVER_ERROR`;
export const GraphqlErrorsExtensionsCodeInvalidInputCustomComponentMapKey = `${GraphqlErrorsExtensionsCodeCustomComponentMapKeyPrefix}INVALID_INPUT`;
export const GraphqlErrorsExtensionsCodeLockedCustomComponentMapKey = `${GraphqlErrorsExtensionsCodeCustomComponentMapKeyPrefix}LOCKED`;
export const GraphqlErrorsExtensionsCodeNotFoundCustomComponentMapKey = `${GraphqlErrorsExtensionsCodeCustomComponentMapKeyPrefix}NOT_FOUND`;
export const GraphqlErrorsExtensionsCodeTooManyRequestsCustomComponentMapKey = `${GraphqlErrorsExtensionsCodeCustomComponentMapKeyPrefix}TOO_MANY_REQUESTS`;
export const GraphqlErrorsExtensionsCodeUnauthorizedCustomComponentMapKey = `${GraphqlErrorsExtensionsCodeCustomComponentMapKeyPrefix}UNAUTHORIZED`;
