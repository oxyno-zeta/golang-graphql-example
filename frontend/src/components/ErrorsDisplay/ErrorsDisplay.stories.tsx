import React from 'react';
import { type StoryFn, type Meta } from '@storybook/react-vite';
import { CombinedGraphQLErrors } from '@apollo/client';
import ErrorsDisplay, { type Props } from './ErrorsDisplay';
import {
  forbiddenNetworkError,
  simpleCombinedGraphQLErrorWithoutExtension,
  simpleForbiddenCombinedGraphQLError,
  simpleForbiddenGraphqlError,
  simpleInternalServerErrorGraphqlError,
  simpleInternalServerWithTraceErrorCombinedGraphQLError,
} from './ErrorsDisplay.storage-test';

export default {
  title: 'Components/ErrorsDisplay',
  component: ErrorsDisplay,
  args: {
    error: forbiddenNetworkError,
  },
} as Meta<typeof ErrorsDisplay>;

export const Playground: StoryFn<typeof ErrorsDisplay> = function C(args: Props) {
  return <ErrorsDisplay {...args} />;
};

export const ClassicError: StoryFn<typeof ErrorsDisplay> = function C() {
  return <ErrorsDisplay error={new Error('fake error !')} />;
};

export const GraphQLNetworkError: StoryFn<typeof ErrorsDisplay> = function C() {
  return <ErrorsDisplay error={forbiddenNetworkError} />;
};

export const OneGraphQLErrorWithoutExtension: StoryFn<typeof ErrorsDisplay> = function C() {
  return <ErrorsDisplay error={simpleCombinedGraphQLErrorWithoutExtension} />;
};

export const OneGraphQLErrorWithExtension: StoryFn<typeof ErrorsDisplay> = function C() {
  return <ErrorsDisplay error={simpleForbiddenCombinedGraphQLError} />;
};

export const TwoGraphQLErrorWithExtension: StoryFn<typeof ErrorsDisplay> = function C() {
  return (
    <ErrorsDisplay
      error={
        new CombinedGraphQLErrors({
          errors: [simpleForbiddenGraphqlError, simpleInternalServerErrorGraphqlError],
        })
      }
    />
  );
};

export const OneGraphQLErrorWithoutMargin: StoryFn<typeof ErrorsDisplay> = function C() {
  return <ErrorsDisplay error={simpleForbiddenCombinedGraphQLError} noMargin />;
};

export const OneWithTraceGraphQLError: StoryFn<typeof ErrorsDisplay> = function C() {
  return <ErrorsDisplay error={simpleInternalServerWithTraceErrorCombinedGraphQLError} />;
};
