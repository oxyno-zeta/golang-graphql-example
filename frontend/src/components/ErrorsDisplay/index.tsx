import ErrorsDisplay, { Props as ErrorsDisplayProps } from './ErrorsDisplay';

export default ErrorsDisplay;
export type { ErrorsDisplayProps };
export * from './constants';
