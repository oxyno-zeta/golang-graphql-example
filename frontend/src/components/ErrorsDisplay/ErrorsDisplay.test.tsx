import React from 'react';
import { render } from '@testing-library/react';
// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import '@testing-library/jest-dom';
import { CombinedGraphQLErrors } from '@apollo/client';
import {
  forbiddenNetworkError,
  simpleForbiddenCombinedGraphQLError,
  simpleForbiddenGraphqlError,
  simpleGraphqlErrorWithoutExtension,
  simpleInternalServerErrorCombinedGraphQLError,
  simpleInternalServerErrorGraphqlError,
} from './ErrorsDisplay.storage-test';

import ErrorsDisplay from './ErrorsDisplay';
import { GraphqlErrorsExtensionsCodeForbiddenCustomComponentMapKey } from './constants';

jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key: string) => key }),
}));

describe('ErrorsDisplay', () => {
  it('should return null when no error or errors are present', () => {
    const { container } = render(<ErrorsDisplay />);

    expect(container).toMatchSnapshot();
  });

  it('should display a network error when error is present', () => {
    const { container } = render(<ErrorsDisplay error={forbiddenNetworkError} />);

    const allP = container.querySelectorAll('p');
    expect(allP).toHaveLength(2);

    // Prepare values
    const values = ['common.errors:', 'Forbidden'];
    values.forEach((item) => {
      expect(container).toHaveTextContent(item);
    });

    expect(container).toMatchSnapshot();
  });

  it('should display a graphql error without extension when error is present', () => {
    const { container } = render(<ErrorsDisplay error={simpleGraphqlErrorWithoutExtension} />);

    const allP = container.querySelectorAll('p');
    expect(allP).toHaveLength(2);

    // Prepare values
    const values = ['common.errors:', 'simple graphql error'];
    values.forEach((item) => {
      expect(container).toHaveTextContent(item);
    });

    expect(container).toMatchSnapshot();
  });

  it('should display a graphql error without extension when errors are present', () => {
    const { container } = render(<ErrorsDisplay errors={[simpleGraphqlErrorWithoutExtension]} />);

    const allP = container.querySelectorAll('p');
    expect(allP).toHaveLength(2);

    // Prepare values
    const values = ['common.errors:', 'simple graphql error'];
    values.forEach((item) => {
      expect(container).toHaveTextContent(item);
    });

    expect(container).toMatchSnapshot();
  });

  it('should display a graphql error with extension when error is present', () => {
    const { container } = render(<ErrorsDisplay error={simpleForbiddenCombinedGraphQLError} />);

    const allP = container.querySelectorAll('p');
    expect(allP).toHaveLength(2);

    // Prepare values
    const values = ['common.errors:', 'common.errorCode.FORBIDDEN'];
    values.forEach((item) => {
      expect(container).toHaveTextContent(item);
    });

    expect(container).toMatchSnapshot();
  });

  it('should display a graphql error with extension when errors are present', () => {
    const { container } = render(<ErrorsDisplay errors={[simpleForbiddenCombinedGraphQLError]} />);

    const allP = container.querySelectorAll('p');
    expect(allP).toHaveLength(2);

    // Prepare values
    const values = ['common.errors:', 'common.errorCode.FORBIDDEN'];
    values.forEach((item) => {
      expect(container).toHaveTextContent(item);
    });

    expect(container).toMatchSnapshot();
  });

  it('should display two graphql error with extension when error is present', () => {
    const { container } = render(
      <ErrorsDisplay
        error={
          new CombinedGraphQLErrors({ errors: [simpleForbiddenGraphqlError, simpleInternalServerErrorGraphqlError] })
        }
      />,
    );

    const allP = container.querySelectorAll('p');
    expect(allP).toHaveLength(3);

    // Prepare values
    const values = ['common.errors:', 'common.errorCode.FORBIDDEN', 'common.errorCode.INTERNAL_SERVER_ERROR'];
    values.forEach((item) => {
      expect(container).toHaveTextContent(item);
    });

    expect(container).toMatchSnapshot();
  });

  it('should display two graphql error with extension when errors are present (1 item with 2)', () => {
    const { container } = render(
      <ErrorsDisplay
        errors={[
          new CombinedGraphQLErrors({
            errors: [simpleForbiddenGraphqlError, simpleInternalServerErrorGraphqlError],
          }),
        ]}
      />,
    );

    const allP = container.querySelectorAll('p');
    expect(allP).toHaveLength(3);

    // Prepare values
    const values = ['common.errors:', 'common.errorCode.FORBIDDEN', 'common.errorCode.INTERNAL_SERVER_ERROR'];
    values.forEach((item) => {
      expect(container).toHaveTextContent(item);
    });

    expect(container).toMatchSnapshot();
  });

  it('should display two graphql error with extension when errors are present (2 items)', () => {
    const { container } = render(
      <ErrorsDisplay errors={[simpleForbiddenCombinedGraphQLError, simpleInternalServerErrorCombinedGraphQLError]} />,
    );

    const allP = container.querySelectorAll('p');
    expect(allP).toHaveLength(3);

    // Prepare values
    const values = ['common.errors:', 'common.errorCode.FORBIDDEN', 'common.errorCode.INTERNAL_SERVER_ERROR'];
    values.forEach((item) => {
      expect(container).toHaveTextContent(item);
    });

    expect(container).toMatchSnapshot();
  });

  it('should display two graphql error with extension when errors are present (2 items with 1 custom component)', () => {
    function Fake({ input }: { readonly input: string }) {
      return <p>{input}</p>;
    }

    const { container } = render(
      <ErrorsDisplay
        customErrorComponentProps={{
          [GraphqlErrorsExtensionsCodeForbiddenCustomComponentMapKey]: { input: 'fake' },
        }}
        customErrorComponents={{
          [GraphqlErrorsExtensionsCodeForbiddenCustomComponentMapKey]: Fake,
        }}
        errors={[simpleForbiddenCombinedGraphQLError, simpleInternalServerErrorCombinedGraphQLError]}
      />,
    );

    const allP = container.querySelectorAll('p');
    expect(allP).toHaveLength(3);

    // Prepare values
    const values = ['common.errors:', 'fake', 'common.errorCode.INTERNAL_SERVER_ERROR'];
    values.forEach((item) => {
      expect(container).toHaveTextContent(item);
    });
    expect(container).not.toHaveTextContent('common.errorCode.FORBIDDEN');

    expect(container).toMatchSnapshot();
  });

  it('should display classic error when error is present', () => {
    const { container } = render(<ErrorsDisplay error={new Error('error1')} />);

    const allP = container.querySelectorAll('p');
    expect(allP).toHaveLength(2);

    // Prepare values
    const values = ['common.errors:', 'error1'];
    values.forEach((item) => {
      expect(container).toHaveTextContent(item);
    });

    expect(container).toMatchSnapshot();
  });

  it('should display classic error when errors are present (1 item)', () => {
    const { container } = render(<ErrorsDisplay errors={[new Error('error1')]} />);

    const allP = container.querySelectorAll('p');
    expect(allP).toHaveLength(2);

    // Prepare values
    const values = ['common.errors:', 'error1'];
    values.forEach((item) => {
      expect(container).toHaveTextContent(item);
    });

    expect(container).toMatchSnapshot();
  });

  it('should display classic error when errors are present (2 items)', () => {
    const { container } = render(<ErrorsDisplay errors={[new Error('error1'), new Error('error2')]} />);

    const allP = container.querySelectorAll('p');
    expect(allP).toHaveLength(3);

    // Prepare values
    const values = ['common.errors:', 'error1', 'error2'];
    values.forEach((item) => {
      expect(container).toHaveTextContent(item);
    });

    expect(container).toMatchSnapshot();
  });
});
