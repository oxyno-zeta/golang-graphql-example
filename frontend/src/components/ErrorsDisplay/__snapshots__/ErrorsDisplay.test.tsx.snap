// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ErrorsDisplay should display a graphql error with extension and trace and request id when with trace error is present 1`] = `
.emotion-0 {
  margin: 64px;
}

.emotion-0 color {
  main: #d32f2f;
  light: #ef5350;
  dark: #c62828;
  contrast-text: #fff;
}

.emotion-1 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
  color: #d32f2f;
}

.emotion-5 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 0.75rem;
  line-height: 1.66;
  letter-spacing: 0.03333em;
  color: #d32f2f;
}

<div>
  <div
    class="MuiBox-root emotion-0"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 emotion-1"
    >
      common.errors
      :
    </p>
    <ul
      class="MuiBox-root emotion-2"
    >
      <li
        class="MuiBox-root emotion-2"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 emotion-1"
        >
          common.errorCode.INTERNAL_SERVER_ERROR
        </p>
        <p
          class="MuiTypography-root MuiTypography-caption emotion-5"
        >
          common.supportTraceId
          : 
          trace-id
        </p>
        <p
          class="MuiTypography-root MuiTypography-caption emotion-5"
        >
          common.supportRequestId
          : 
          request-id
        </p>
      </li>
    </ul>
  </div>
</div>
`;

exports[`ErrorsDisplay should display a graphql error with extension when error is present 1`] = `
.emotion-0 {
  margin: 64px;
}

.emotion-0 color {
  main: #d32f2f;
  light: #ef5350;
  dark: #c62828;
  contrast-text: #fff;
}

.emotion-1 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
  color: #d32f2f;
}

<div>
  <div
    class="MuiBox-root emotion-0"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 emotion-1"
    >
      common.errors
      :
    </p>
    <ul
      class="MuiBox-root emotion-2"
    >
      <li
        class="MuiBox-root emotion-2"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 emotion-1"
        >
          common.errorCode.FORBIDDEN
        </p>
      </li>
    </ul>
  </div>
</div>
`;

exports[`ErrorsDisplay should display a graphql error with extension when errors are present 1`] = `
.emotion-0 {
  margin: 64px;
}

.emotion-0 color {
  main: #d32f2f;
  light: #ef5350;
  dark: #c62828;
  contrast-text: #fff;
}

.emotion-1 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
  color: #d32f2f;
}

<div>
  <div
    class="MuiBox-root emotion-0"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 emotion-1"
    >
      common.errors
      :
    </p>
    <ul
      class="MuiBox-root emotion-2"
    >
      <li
        class="MuiBox-root emotion-2"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 emotion-1"
        >
          common.errorCode.FORBIDDEN
        </p>
      </li>
    </ul>
  </div>
</div>
`;

exports[`ErrorsDisplay should display a graphql error without extension when error is present 1`] = `
.emotion-0 {
  margin: 64px;
}

.emotion-0 color {
  main: #d32f2f;
  light: #ef5350;
  dark: #c62828;
  contrast-text: #fff;
}

.emotion-1 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
  color: #d32f2f;
}

<div>
  <div
    class="MuiBox-root emotion-0"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 emotion-1"
    >
      common.errors
      :
    </p>
    <ul
      class="MuiBox-root emotion-2"
    >
      <li
        class="MuiBox-root emotion-2"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 emotion-1"
        >
          simple graphql error
        </p>
      </li>
    </ul>
  </div>
</div>
`;

exports[`ErrorsDisplay should display a graphql error without extension when errors are present 1`] = `
.emotion-0 {
  margin: 64px;
}

.emotion-0 color {
  main: #d32f2f;
  light: #ef5350;
  dark: #c62828;
  contrast-text: #fff;
}

.emotion-1 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
  color: #d32f2f;
}

<div>
  <div
    class="MuiBox-root emotion-0"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 emotion-1"
    >
      common.errors
      :
    </p>
    <ul
      class="MuiBox-root emotion-2"
    >
      <li
        class="MuiBox-root emotion-2"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 emotion-1"
        >
          simple graphql error
        </p>
      </li>
    </ul>
  </div>
</div>
`;

exports[`ErrorsDisplay should display a network error when error is present 1`] = `
.emotion-0 {
  margin: 64px;
}

.emotion-0 color {
  main: #d32f2f;
  light: #ef5350;
  dark: #c62828;
  contrast-text: #fff;
}

.emotion-1 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
  color: #d32f2f;
}

<div>
  <div
    class="MuiBox-root emotion-0"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 emotion-1"
    >
      common.errors
      :
    </p>
    <ul
      class="MuiBox-root emotion-2"
    >
      <li
        class="MuiBox-root emotion-2"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 emotion-1"
        >
          Forbidden
        </p>
      </li>
    </ul>
  </div>
</div>
`;

exports[`ErrorsDisplay should display classic error when error is present 1`] = `
.emotion-0 {
  margin: 64px;
}

.emotion-0 color {
  main: #d32f2f;
  light: #ef5350;
  dark: #c62828;
  contrast-text: #fff;
}

.emotion-1 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
  color: #d32f2f;
}

<div>
  <div
    class="MuiBox-root emotion-0"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 emotion-1"
    >
      common.errors
      :
    </p>
    <ul
      class="MuiBox-root emotion-2"
    >
      <li
        class="MuiBox-root emotion-2"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 emotion-1"
        >
          error1
        </p>
      </li>
    </ul>
  </div>
</div>
`;

exports[`ErrorsDisplay should display classic error when errors are present (1 item) 1`] = `
.emotion-0 {
  margin: 64px;
}

.emotion-0 color {
  main: #d32f2f;
  light: #ef5350;
  dark: #c62828;
  contrast-text: #fff;
}

.emotion-1 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
  color: #d32f2f;
}

<div>
  <div
    class="MuiBox-root emotion-0"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 emotion-1"
    >
      common.errors
      :
    </p>
    <ul
      class="MuiBox-root emotion-2"
    >
      <li
        class="MuiBox-root emotion-2"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 emotion-1"
        >
          error1
        </p>
      </li>
    </ul>
  </div>
</div>
`;

exports[`ErrorsDisplay should display classic error when errors are present (2 items) 1`] = `
.emotion-0 {
  margin: 64px;
}

.emotion-0 color {
  main: #d32f2f;
  light: #ef5350;
  dark: #c62828;
  contrast-text: #fff;
}

.emotion-1 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
  color: #d32f2f;
}

<div>
  <div
    class="MuiBox-root emotion-0"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 emotion-1"
    >
      common.errors
      :
    </p>
    <ul
      class="MuiBox-root emotion-2"
    >
      <li
        class="MuiBox-root emotion-2"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 emotion-1"
        >
          error1
        </p>
      </li>
      <li
        class="MuiBox-root emotion-2"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 emotion-1"
        >
          error2
        </p>
      </li>
    </ul>
  </div>
</div>
`;

exports[`ErrorsDisplay should display two graphql error with extension when error is present 1`] = `
.emotion-0 {
  margin: 64px;
}

.emotion-0 color {
  main: #d32f2f;
  light: #ef5350;
  dark: #c62828;
  contrast-text: #fff;
}

.emotion-1 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
  color: #d32f2f;
}

<div>
  <div
    class="MuiBox-root emotion-0"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 emotion-1"
    >
      common.errors
      :
    </p>
    <ul
      class="MuiBox-root emotion-2"
    >
      <li
        class="MuiBox-root emotion-2"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 emotion-1"
        >
          common.errorCode.FORBIDDEN
        </p>
      </li>
      <li
        class="MuiBox-root emotion-2"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 emotion-1"
        >
          common.errorCode.INTERNAL_SERVER_ERROR
        </p>
      </li>
    </ul>
  </div>
</div>
`;

exports[`ErrorsDisplay should display two graphql error with extension when errors are present (1 item with 2) 1`] = `
.emotion-0 {
  margin: 64px;
}

.emotion-0 color {
  main: #d32f2f;
  light: #ef5350;
  dark: #c62828;
  contrast-text: #fff;
}

.emotion-1 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
  color: #d32f2f;
}

<div>
  <div
    class="MuiBox-root emotion-0"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 emotion-1"
    >
      common.errors
      :
    </p>
    <ul
      class="MuiBox-root emotion-2"
    >
      <li
        class="MuiBox-root emotion-2"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 emotion-1"
        >
          common.errorCode.FORBIDDEN
        </p>
      </li>
      <li
        class="MuiBox-root emotion-2"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 emotion-1"
        >
          common.errorCode.INTERNAL_SERVER_ERROR
        </p>
      </li>
    </ul>
  </div>
</div>
`;

exports[`ErrorsDisplay should display two graphql error with extension when errors are present (2 items and 1 with trace error) 1`] = `
.emotion-0 {
  margin: 64px;
}

.emotion-0 color {
  main: #d32f2f;
  light: #ef5350;
  dark: #c62828;
  contrast-text: #fff;
}

.emotion-1 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
  color: #d32f2f;
}

.emotion-7 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 0.75rem;
  line-height: 1.66;
  letter-spacing: 0.03333em;
  color: #d32f2f;
}

<div>
  <div
    class="MuiBox-root emotion-0"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 emotion-1"
    >
      common.errors
      :
    </p>
    <ul
      class="MuiBox-root emotion-2"
    >
      <li
        class="MuiBox-root emotion-2"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 emotion-1"
        >
          common.errorCode.FORBIDDEN
        </p>
      </li>
      <li
        class="MuiBox-root emotion-2"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 emotion-1"
        >
          common.errorCode.INTERNAL_SERVER_ERROR
        </p>
        <p
          class="MuiTypography-root MuiTypography-caption emotion-7"
        >
          common.supportTraceId
          : 
          trace-id
        </p>
        <p
          class="MuiTypography-root MuiTypography-caption emotion-7"
        >
          common.supportRequestId
          : 
          request-id
        </p>
      </li>
    </ul>
  </div>
</div>
`;

exports[`ErrorsDisplay should display two graphql error with extension when errors are present (2 items with 1 custom component) 1`] = `
.emotion-0 {
  margin: 64px;
}

.emotion-0 color {
  main: #d32f2f;
  light: #ef5350;
  dark: #c62828;
  contrast-text: #fff;
}

.emotion-1 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
  color: #d32f2f;
}

<div>
  <div
    class="MuiBox-root emotion-0"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 emotion-1"
    >
      common.errors
      :
    </p>
    <ul
      class="MuiBox-root emotion-2"
    >
      <li
        class="MuiBox-root emotion-2"
      >
        <p>
          fake
        </p>
      </li>
      <li
        class="MuiBox-root emotion-2"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 emotion-1"
        >
          common.errorCode.INTERNAL_SERVER_ERROR
        </p>
      </li>
    </ul>
  </div>
</div>
`;

exports[`ErrorsDisplay should display two graphql error with extension when errors are present (2 items) 1`] = `
.emotion-0 {
  margin: 64px;
}

.emotion-0 color {
  main: #d32f2f;
  light: #ef5350;
  dark: #c62828;
  contrast-text: #fff;
}

.emotion-1 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
  color: #d32f2f;
}

<div>
  <div
    class="MuiBox-root emotion-0"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 emotion-1"
    >
      common.errors
      :
    </p>
    <ul
      class="MuiBox-root emotion-2"
    >
      <li
        class="MuiBox-root emotion-2"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 emotion-1"
        >
          common.errorCode.FORBIDDEN
        </p>
      </li>
      <li
        class="MuiBox-root emotion-2"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 emotion-1"
        >
          common.errorCode.INTERNAL_SERVER_ERROR
        </p>
      </li>
    </ul>
  </div>
</div>
`;

exports[`ErrorsDisplay should return null when no error or errors are present 1`] = `<div />`;
